src/hrg/hrg.cc

```diff
@@ -285,6 +285,12 @@ static igraph_error_t igraph_i_hrg_getsimplegraph(const igraph_t *igraph,
  */
 
 igraph_error_t igraph_hrg_init(igraph_hrg_t *hrg, igraph_integer_t n) {
+    if (n < 0) {
+        IGRAPH_ERRORF("Number of vertices should be non-negative, found %" IGRAPH_PRId ".", IGRAPH_EINVAL, n);
```

Comment:
Nitpick: Let use the friendlier phrasing "should not be negative" for these cases.